### PR TITLE
ci: move the Jenkins Job Builder .yaml files under jobs/

### PR DESCRIPTION
**Pull request by nixpanic**
# Describe what this PR does #

There is a new .commitlintrc.yml file in the root of the repository.
This causes jenkins-job-builder to complains that the file is not
confirm the Jenkins Job Builder format/specification.

    jenkins_jobs.errors.JenkinsJobsException: The topmost collection in file '/opt/build/.commitlintrc.yml' must be a list, not a <class 'collections.OrderedDict'>

Move all Jenkins Job Builder .yaml files to jobs/ so that all the
Jenkins job configurations are kept together without other files.

### CODE REVIEW

**Comment by nixpanic** [comment]
@Yuggupta27 please review this, and place any new `e2e.yaml` files in the `jobs/` directory.

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ projects branches.
 As an example, the `containerized-tests` Jenkins job consists out of the
 following files:
 
-- `containerized-tests.yaml` is a [Jenkins Job Builder](jjb) configuration
+- `jobs/containerized-tests.yaml` is a [Jenkins Job Builder](jjb) configuration
```
Comment (Madhu-1) [inline README.md:19]: [Jenkins Job Builder](jjb) is not linked to any file when I open the link its throwing 404 error, this needs to be changed isn't it

**Comment by Madhu-1** [comment]
https://github.com/ceph/ceph-csi/blob/bcacb641bdd69add77f15ab907bfb9dce8656f28/pipeline
https://github.com/ceph/ceph-csi/blob/bcacb641bdd69add77f15ab907bfb9dce8656f28/centos_ci
https://github.com/ceph/ceph-csi/blob/bcacb641bdd69add77f15ab907bfb9dce8656f28/jjb
broken links

File: README.md (hunk 1 above)
Comment (nixpanic) [reply README.md:19]: Oh, indeed! The linking is broken, it is `[title](id)` now, but should be `[title][id]` instead. I'll send an additional commit to this PR to fix it.

**Comment by nixpanic** [comment]
The broken links have been addressed now too.

File: README.md [hunk 2]
```diff
@@ -16,12 +16,12 @@ projects branches.
...
 
-- `containerized-tests.yaml` is a [Jenkins Job Builder](jjb) configuration
+- `jobs/containerized-tests.yaml` is a [Jenkins Job Builder][jjb] configuration
   that describes the events when the job should get run and fetches the
   `.groovy` file from the git repository/branch
-- `containerized-tests.groovy` is the [Jenkins Pipeline](pipeline) that
+- `containerized-tests.groovy` is the [Jenkins Pipeline][pipeline] that
   contains the stages for the Jenkins Job itself. In order to work with [the
-  bare-metal machines from the CentOS CI](centos_ci), it executes the following
+  bare-metal machines from the CentOS CI][centos_ci], it executes the following
```
Comment (Madhu-1) [inline README.md:24]: the link is not generated here

Comment (Yuggupta27) [reply README.md:24]: I guess the bracket pattern will be the same as the previous version @nixpanic

Comment (nixpanic) [reply README.md]: `[centos_ci]` is not a valkid id, `[centos_ci_hw]` is listed at the bottom of the file

**Comment by nixpanic** [comment]
Fixed again, please check :merperson:

**Comment by Yuggupta27** [comment]
> @Yuggupta27 please review this, and place any new `e2e.yaml` files in the `jobs/` directory.

Sure.

**Pull request merged**